### ISSUE TITLE
Fabric "needs to prompt" in parallel mode, but not in serial mode

### ISSUE THREAD

**Issue by nhooey** [fabric/fabric#1433]
I'm getting this error when running a Fabric task in parallel mode:

```
No handlers could be found for logger "paramiko.transport"

Fatal error: Needed to prompt for a connection or sudo password (host: node01 node02),
but input would be ambiguous in parallel mode

Aborting.
[node01 node02] run: sudo supervisorctl stop whatever
```

This is with the following code:
```python
from fabric.api import execute, parallel, run, task

env.use_ssh_config = True
env.use_shell = False

@task
@parallel(pool_size=2)
def deploy():
    run('sudo supervisorctl stop %s' % supervisor_job)

deploy.hosts = ['node01', 'node02']
execute(deploy)
```

However, if I turn off parallel, I don't see any SSH password or sudo prompts, or any errors about them:
```python
@parallel(pool_size=1)
```

I can also confirm that there are no prompts when SSHing to the `node01` and `node02` boxes manually, nor are there any sudo prompts when running the exact commands on the boxes.

This seems to only be an issue with parallel mode.

I've had so many issues with password prompts that I think it would be a good idea to have global settings for disabling both SSH password and sudo prompts. Fabric seems to guess that the prompts are necessary, and breaks when there is no standard input (when using Atlassian Bamboo, Jenkins, or some other build system).

I'd prefer to just manually tell Fabric never to prompt for anything, and fail if the operating system actually prompts.

**Comment by nhooey** [comment]
Okay this issue is completely off.

Someone changed the list of hosts in some environment variable such that one of the hosts was called `node01 node02` with a space in it, but it wasn't obvious.

Still, Fabric shouldn't say that it's prompting for a password when it couldn't possibly SSH to the host.

**Comment by v6** [comment]
// , Fabric isn't really designed to run in Parallel mode, and is not thread safe in it. Consider a saner library if you need this sort of thing

**Comment by nhooey** [comment]
Yeah, Fabric 1 is pretty sloppy with how it runs. Fabric 2 is supposed to fix a lot of these issues.

I actually had good success with using the Ansible Python API. The documentation sucked at the time, but I was able to write a concurrent build system with it.

What other libraries would you suggest?

**Comment by v6** [comment]
// , I think ParallelSSH was designed to solve the problems you mentioned above: 

![image](https://cloud.githubusercontent.com/assets/5750764/23075659/2862069a-f4f2-11e6-8f23-f237aef3fb7f.png)

https://github.com/ParallelSSH/parallel-ssh#frequently-asked-questions
https://github.com/ParallelSSH/parallel-ssh#design-and-goals
https://github.com/ParallelSSH/parallel-ssh#installation

**Comment by v6** [comment]
// , Fabric was mostly intended as a faithful port of Capistrano.
